src/app.js

```diff
@@ -1,3 +1,36 @@
+/* eslint-disable no-console */
 'use strict';
 
-// Write your code here
+const chalk = require('chalk');
+const readline = require('readline');
+const { checkIsValidUserInput } = require('./modules/checkIsValidUserInput');
+const { generateRandomNumber } = require('./modules/generateRandomNumber');
+const { getBullsAndCows } = require('./modules/getBullsAndCows');
+
+const terminal = readline.createInterface(
+  process.stdin,
+  process.stdout
+);
+
+const randomNumber = generateRandomNumber();
+
+const main = () => {
+  terminal.question('Enter number: ', (input) => {
+    if (!checkIsValidUserInput(input)) {
+      console.log(chalk.red('Number is not valid. Try again.'));
+      main();
+    } else {
+      const result = getBullsAndCows(input, randomNumber);
+
+      if (result.bulls !== 4) {
+        console.log(`${JSON.stringify(result)}`, chalk.yellow('Guess again'));
+        main();
+      } else {
+        console.log(chalk.green('Congratulations! You guessed the number!'));
```

Comment:
Consider adding a condition to handle if the input is not a number, for better error handling.